Core/Src/main.c

```diff
@@ -95,10 +103,22 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //setting the PB4 port (I think that's the GPIO port?) low
+
+	  HAL_SPI_TransmitReceive(&hspi1, T_buffer, R_buffer, 8, 500);
+	  adc_value[1] = (R_buffer[2] & 0x03) + R_buffer[3]; // masking with 0x03 (0000 0011) only extracts the last two digit.
```

Comment:
I definitely see what you are trying to do here and it is a common misconception when dealing with bit manipulations. There are a couple of issues with line 109.

1) This is syntax related: I see that you are trying to get a value from `R_buffer[3]` but that value is out of range. What I mean is `R_Buffer` was declared as a size of 3 meaning it has indexes 0-2 (a total of 3 indexes), so there does not exist an 'R_buffer[3]'

2) The masking with 0x03 although valid, is not necessary in THIS specific scenario because all the values in front of the important bits are 0, so they have no impact on the value. If, for example, the bits in front of the ones we were looking at were random values, then YES we would have to mask them with 0x03.

3) I completely understand what you mean by adding the two parts together but that will yield an incorrect result. In this case, don't think of the two different values as something you have to 'add' together but something you have to 'stitch' together.

Let me give you an example.

If the two values you were looking at were 0000 0011 and 0100 1101, here is what I mean by stitching those two values together: 0000 0011 0100 1101

NOTE: the two 8 bit values were stitched together to create a 16 bit value (it looks like you already have `adc_value` as a 16 bit value, so that's great!

Here I am literally putting the two different values together, NOT adding them because adding the values would yield a result of 01010000 (an 8 bit value NOT 16 bit) which is not even close to the value I want.

To get the desired result of 0000 0011 0100 1101 it will require BITWISE operations, specifically the `<<` and `|` operator. I will only tell you which two operators you will require, but not how to use them. It is up to you to determine how to use them in this situation!